### ISSUE TITLE
Sequential concatentation to start before the previous sound has finished

### ISSUE THREAD

**Issue by ThomasVeitch** [ZeroOneZeroR/android_audio_mixer#3]
Not really an issue but a requirement.

It would be useful if you were able to make a wav file from many wav files. But instead of sequentially waiting until one sound finished before concatenating it to the end, you can start a sound before the first finished. So really I'm looking for a mixture of parallel and sequential, but the sequential being able to be added at a specific time before the previous sound has finished.
Is this possible?
Could you do it?

Thanks.

**Comment by ZeroOneZeroR** [comment]
What you want can be done by this. You just need to use parallel mixing with blank audio input.

Aren't wav files supported?

**Comment by ThomasVeitch** [comment]
Thanks for getting back to me quickly,

You'll have to detail this as I've tried

audioMixer.addDataSource(input1); // 2 secs long
audioMixer.addDataSource(blankInput); // 0.5 secs gap
audioMixer.addDataSource(input2); // 2 secs long

audioMixer.setMixingType(AudioMixer.MixingType.PARALLEL);


The desired output would be to play at 0 secs input 1; then at 0.5
secs play input 2. (when input 1 is still playing)

The above is just putting everything in parallel so they all play at
the same time.

If I change the setting to SEQUENTIAL then at 0 secs input 1 is
played, then at 2 secs there is a gap at 2.5 secs input 2 is played.

Please explain how to achieve desired output.

Tom

**Comment by ZeroOneZeroR** [comment]
Sorry, I gave the wrong answer.
Currently this not possible with this. Need to do some work.

**Comment by ThomasVeitch** [comment]
Thanks.

I look forward to it!

Tom.

**Comment by ThomasVeitch** [comment]
Hi Rakibul,

When will this be ready? I'd really like this functionality.

Tom

**Comment by ZeroOneZeroR** [comment]
I will work on it soon.

**Comment by ZeroOneZeroR** [comment]
Your problem has been solved. You can now set startOffsetUs() to GeneralAudioInput. Use version v1.1

**Comment by ZeroOneZeroR** [comment]
Do you have the previous version's code? I need the README.md file. Could you provide me with that, please?

**Comment by ThomasVeitch** [comment]
Just did a quick rattle test and it's looking fantastic. So many thanks.
It's looking like an excellent API.

**Comment by ZeroOneZeroR** [comment]
Thank you so much.

**Comment by ThomasVeitch** [comment]
Hi Rakibul,
We are using the API in a proper project and it's become apparent we need
some functionality adding.
We have created about a 10second wav file and we intend to loop it using
MediaPlayer setLooping(true)
The issue is we don't want to wait until the loop has finished to start
again.
We want to start the next loop at a designated time before
the previous loop has finished.
Would this be possible?
Thanks.

**Comment by ZeroOneZeroR** [comment]
You want this while playing with Mediaplayer? I haven't got you properly.

**Comment by ThomasVeitch** [comment]
Yes. Thanks to your API we have created a 15 second .wav file with lots of
individual sounds running in parallel and sequentially.
With the offSet functionality we have been able to mix sounds properly
without having to wait until a previous sound has finished.

However we intend to loop this newly created 15 second .wav file.
We are using MediaPlayer to play this .wav file.

Unfortunately, when trying to loop this new .wav file, you have to wait
until the whole wav file has finished playing.
Are you able to extend your functionality so that we can loop the wav file,
without having to wait until the very end of the file. We need to able to
start the new loop at a designated offset before the previous loop has
finished.

**Comment by ZeroOneZeroR** [comment]
You can add one audio multiple times. You can add the previous Audios again with  new AudioInputs with your required settings. You have to calculate the startOffset for looping.

**Comment by ThomasVeitch** [comment]
The issue with creating a huge wav file with AudioMixer (and not looping in
MediaPlayer) is that it''s slow to create and the volume is really low.

**Comment by ZeroOneZeroR** [comment]
Maybe i will have to do something to maintain the volume. I already have thought about it. Let me do something with volume. But i can't do anything with timing.
